pkg/tsdb/cloudwatch/resource_handler.go

```diff
@@ -0,0 +1,59 @@
+package cloudwatch
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter"
+)
+
+func (e *cloudWatchExecutor) newResourceMux() *http.ServeMux {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/regions", handleResourceReq(e.handleGetRegions))
+	mux.HandleFunc("/namespaces", handleResourceReq(e.handleGetNamespaces))
+	mux.HandleFunc("/metrics", handleResourceReq(e.handleGetMetrics))
+	mux.HandleFunc("/all-metrics", handleResourceReq(e.handleGetAllMetrics))
+	mux.HandleFunc("/dimension-keys", handleResourceReq(e.handleGetDimensionKeys))
+	mux.HandleFunc("/dimension-values", handleResourceReq(e.handleGetDimensionValues))
+	mux.HandleFunc("/ebs-volume-ids", handleResourceReq(e.handleGetEbsVolumeIds))
+	mux.HandleFunc("/ec2-instance-attribute", handleResourceReq(e.handleGetEc2InstanceAttribute))
+	mux.HandleFunc("/resource-arns", handleResourceReq(e.handleGetResourceArns))
+	return mux
+}
+
+type handleFn func(pluginCtx backend.PluginContext, parameters url.Values) ([]suggestData, error)
+
+func handleResourceReq(handleMetricFind handleFn) func(rw http.ResponseWriter, req *http.Request) {
+	return func(rw http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		pluginContext := httpadapter.PluginConfigFromContext(ctx)
+		err := req.ParseForm()
+		if err != nil {
+			writeResponse(rw, http.StatusBadRequest, fmt.Sprintf("unexpected error %v", err))
+		}
+		data, err := handleMetricFind(pluginContext, req.URL.Query())
```

Comment:
I'd avoid the "metricFind" naming so it's not confusing

Reply:
done